core/shared/src/main/scala/fs2/interop/flow/StreamSubscription.scala

```diff
@@ -65,6 +65,7 @@ private[flow] final class StreamSubscription[F[_], A] private (
       def go(s: Stream[F, A]): Pull[F, A, Unit] =
         Pull.eval(F.delay(requests.get())).flatMap { n =>
```

Comment:
If we just do this, could we avoid the new loop condition?
```suggestion
        Pull.eval(F.delay(requests.getAndSet(0))).flatMap { n =>
```